supervisor_checks/check_modules/http.py

```diff
@@ -78,7 +79,17 @@ def _make_http_request(self, host_port, timeout,
             headers['Authorization'] = 'Basic %s' % base64.b64encode(
                 auth_str.encode()).decode()
 
-        connection.request('GET', self._config['url'], headers=headers)
+        if self._config['headers']:
```

Comment:
Can you please make all new config keys optional so this check module does not crash with the old code. I have some code when I'm using check modules framework directly, without CLI tools, I want to make sure they still able to work with old config (that does not contain method/headers/body keys).

Just do self._config.get() rather than self._config[...]

Reply:
with how the code is currently used this operation still should be backwards compatible, 
http_check should call with None values as default for the new flags, so the keys should resolve to None.
But its better to be safer than sorry